x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -54,6 +57,9 @@ public record TestCaseSupplier(String name, List<DataType> types, Supplier<TestC
     implements
         Supplier<TestCaseSupplier.TestCase> {
 
+    public static final Source TEST_SOURCE = new Source(new Location(1, 0), "source");
```

Comment:
Do you think it'd be better to randomize this in the places we call it?

Reply:
Actually probably not - we want it to be consistent in the error message.

Reply:
I would like to autogenerate it, so we can also deserialize the fields with a real source, with real names, and remove synthetic Sources from tests if possible.
There's some minor problems I have to handle first to do that, so I'll try in another PR